.circleci/config.yml

```diff
@@ -5,43 +5,119 @@ version: 2.1
+        time sudo /bin/bash nvidia-downloads/cuda_10.2.89_440.33.01_linux.run --no-drm --silent --driver --toolkit
+        sudo ldconfig /usr/local/cuda/lib64
+        echo "Done installing NVIDIA drivers and CUDA libraries."
+        nvidia-smi
+
+setup_cuda11_libs: &setup_cuda11_libs
   - run:
-      name: Setup CUDA
+      name: Setup CUDA drivers and libraries
       working_directory: ~/
       command: |
         # download and install nvidia drivers, cuda, etc
-        wget --quiet --no-clobber -P ~/nvidia-downloads 'https://s3.amazonaws.com/ossci-linux/nvidia_driver/NVIDIA-Linux-x86_64-440.82.run'
-        time sudo /bin/bash ~/nvidia-downloads/NVIDIA-Linux-x86_64-440.82.run --no-drm -q --ui=none
-        echo "Done installing NVIDIA drivers."
-        pyenv versions
+        wget --quiet --no-clobber -P ~/nvidia-downloads https://developer.download.nvidia.com/compute/cuda/11.3.0/local_installers/cuda_11.3.0_465.19.01_linux.run
+        time sudo /bin/bash nvidia-downloads/cuda_11.3.0_465.19.01_linux.run --no-drm --silent --driver --toolkit
+        sudo ldconfig /usr/local/cuda/lib64
+        echo "Done installing NVIDIA drivers and CUDA libraries."
         nvidia-smi
-        pyenv global 3.7.0
-          
+
+setup_environment: &setup_environment
+  - run:
+      name: Setup virtualenv and tools
+      working_directory: ~/project
+      command: |
+        virtualenv ~/theseus_venv -p /usr/bin/python3.7
+        echo ". ~/theseus_venv/bin/activate" >> $BASH_ENV
+        . ~/theseus_venv/bin/activate
+        pip install --progress-bar off --upgrade pip
+        pip install --progress-bar off --upgrade setuptools
+
+install_torch_cuda10: &install_torch_cuda10
+  - run:
+      name: Install Torch for cuda10
+      working_directory: ~/project
+      command: |
+        pip install --progress-bar off torch==1.10.0+cu102 torchvision==0.11.1+cu102 torchaudio==0.10.0+cu102 -f https://download.pytorch.org/whl/cu102/torch_stable.html
+        python -c 'import torch; print("Torch version:", torch.__version__); assert torch.cuda.is_available()'
+
+install_torch_cuda11: &install_torch_cuda11
+  - run:
+      name: Install Torch for cuda11
+      working_directory: ~/project
+      command: |
+        pip install --progress-bar off torch==1.10.0+cu113 torchvision==0.11.1+cu113 torchaudio==0.10.0+cu113 -f https://download.pytorch.org/whl/cu113/torch_stable.html
+        python -c 'import torch; print("Torch version:", torch.__version__); assert torch.cuda.is_available()'
+
+setup_project: &setup_project
+  - run:
+      name: Setting up project
+      working_directory: ~/project
+      command: |
+        pip install --progress-bar off -e .
+
+run_tests: &run_tests
+  - run:
+      name: Running tests
+      working_directory: ~/project
+      command: |
+        pytest -s theseus/tests/test_theseus_layer.py
```

Comment:
Shouldn't we be running all tests `pytest -s theseus`?

Reply:
Yeah, I figured. We can merge once Luis confirms.

Reply:
Yes, we did this since the full test suite takes some time, and IMO there is not need for all these tests to be run both on CPU and GPU. That being said, we should probably run GPU tests on a large suit of unit tests, but we don't have to decide this in this PR. What's important for this PR is that the functionality provided here is also tested in CI, which I'm not sure it's the case.

Reply:
wait, this PR is only about the CI changes, and of course there are not new tests since there is no new functionality. The new tests are added in subsequent PRs, while the new functionalities are added incrementally.

Reply:
OK, that's good, I missed this, sorry. As I mentioned offline, I only responded to comments, haven't yet looked through the PRs again :)

Reply:
Makes sense. I'll flag this on the code coverage issue so we can tackle it there.

@maurimo, I think in your other PRs (if you haven't already) you will have to add a line 'pytest -s theseus/extlib/tests/' and maybe others to capture places where we do need to test GPU functionality.